From 8edbc6382e7069c06a61634979d5e614c15dceea Mon Sep 17 00:00:00 2001
From: i509VCB <git@i509.me>
Date: Sat, 4 Jul 2020 19:55:33 -0500
Subject: [PATCH] Buildscript Changes for remapping

This is needed since gradle does all kinds of funky stuff which makes SC impossible to be hooked up to our remapper. This includes disabling vanilla, which we don't even use.

diff --git a/build.gradle.kts b/build.gradle.kts
index 1fd5765fa..d821578e9 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -18,7 +18,7 @@ plugins {
     `java-library`
     idea
     eclipse
-    id("net.minecrell.licenser") version "0.4.1"
+    id("net.minecrell.licenser")
 }
 
 
@@ -49,7 +49,7 @@ minecraft {
                 subprojects {
 
                 }
-                parent?.minecraft?.accessTransformer(it)
+                minecraft?.accessTransformer(it)
             }
 }
 
@@ -117,7 +117,7 @@ dependencies {
     minecraft("net.minecraft:$minecraftDep:$minecraftVersion")
 
     // api
-    api(project(":SpongeAPI"))
+    api(project("SpongeAPI"))
     api("org.spongepowered:plugin-spi:0.1.1-SNAPSHOT")
 
     // Database stuffs... likely needs to be looked at
@@ -132,7 +132,7 @@ dependencies {
     implementation("org.ow2.asm:asm-tree:6.2")
 
     // Launch Dependencies - Needed to bootstrap the engine(s)
-    launchConfig(project(":SpongeAPI"))
+    launchConfig(project("SpongeAPI"))
     launchConfig("org.spongepowered:plugin-spi:0.1.1-SNAPSHOT")
     launchConfig("org.spongepowered:mixin:0.8")
     launchConfig("org.checkerframework:checker-qual:2.8.1")
@@ -162,10 +162,10 @@ dependencies {
     mixinsConfig(sourceSets["main"].output)
     add(accessors.get().implementationConfigurationName, accessorsConfig)
     add(mixins.get().implementationConfigurationName, mixinsConfig)
-    add(mixins.get().implementationConfigurationName, project(":SpongeAPI"))
+    add(mixins.get().implementationConfigurationName, project("SpongeAPI"))
 
     // Invalid
-    add(invalid.get().implementationConfigurationName, project(":SpongeAPI"))
+    add(invalid.get().implementationConfigurationName, project("SpongeAPI"))
     add(invalid.get().implementationConfigurationName, mixinsConfig)
 }
 
@@ -292,6 +292,7 @@ if (testplugins != null) {
     }
 }
 
+/*
 project("SpongeVanilla") {
     val vanillaProject = this
     apply {
@@ -466,8 +467,10 @@ project("SpongeVanilla") {
             this["url"] = projectUrl
         }
         header = apiProject.file("HEADER.txt")
-
-        include("**/*.java")
+*/
+        //include("**/*.java")
+        /*
         newLine = false
     }
 }
+*/
diff --git a/settings.gradle.kts b/settings.gradle.kts
index 528308be0..98cb46d10 100644
--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -2,7 +2,7 @@ rootProject.name = "SpongeCommon"
 
 include("SpongeAPI")
 include(":SpongeVanilla")
-project(":SpongeVanilla").projectDir = file("vanilla")
+//project(":SpongeVanilla").projectDir = file("vanilla")
 pluginManagement {
     repositories {
         mavenLocal()
-- 
2.25.1.windows.1

